### PR TITLE
MINOR: [CI] Avoid docker-compose on Azure

### PR DESCRIPTION
**Pull request by pitrou**
### Rationale for this change

Same as PR #40949, but for Crossbow builds hosted on Azure Pipelines.

### Are these changes tested?

Yes, they fix CI tests.

### Are there any user-facing changes?

No.

### CODE REVIEW

**Comment by pitrou** [comment]
@github-actions crossbow submit conda-clean test-debian-12-python-3-amd64

**Comment by pitrou** [comment]
@assignUser @kou Can you advise?

File: .github/workflows/archery.yml [hunk 1]
```diff
@@ -61,7 +61,7 @@ jobs:
       - name: Setup Python
         uses: actions/setup-python@v5.1.0
         with:
-          python-version: '3.12'
+          python-version: '3.9'
```
Comment (pitrou) [inline .github/workflows/archery.yml:64]: Required for the docker-compose dependencies to install properly.

**Review by assignUser — APPROVED** [review]
Thanks for getting everything going again! Sorry for the late review.

**Pull request merged**